### DISCUSSION TITLE
How to get nvim-tree to use the same (single) instance in all tabs?

### DISCUSSION THREAD

**Discussion by no-more-secrets** [nvim-tree/nvim-tree.lua#2244] [Q&A] [answered]
Hi, ideally, when I open nvim-tree, I'd like it to be open in all tabs, and to be the same instance (buffer?) when viewed from any tab, basically simulating a persistent file tree view that you'd have in a dedicated IDE.  Is there a way to do that with nvim-tree?

**Comment by eyalk11** [comment]
Why wont you solve it in other way, I have the \of mapping that fuzzy search buffers and adds an existing  buffer as a splited-view .  https://github.com/eyalk11/.vim

**Comment by alex-courtis** [comment, marked as answer]
Tracking at
- #2255

**Comment by mikew** [comment]
I have an example of this using my (not ready for prime-time) nvim-drawer: https://github.com/mikew/vimrc/tree/265cfd7f3d46971ce5f0af9d9f86c37420b9ee48/src/nvim/nvim-drawer#nvim-tree


https://github.com/user-attachments/assets/c7e50f2b-9190-4e80-9475-90ba285cbd51

**Reply by alex-courtis** [reply]
Wow... that will be really useful.
1. Any thoughts on using `winfixbuf`? That's in the future for nvim-tree when the minimum version is 0.10 #2724 
2. Is tiling a possibility e.g. tree at the bottom, aerial above, another tree above

**Reply by mikew** [reply]
1. Hadn't heard of it but I'll check it out!
2. Technically yes, but not sure how to do it with nvim-tree.

WRT point 2, the drawer system itself can handle multiple drawers. Each drawer is given a unique prefix, that's how they're found / restored / saved (there's also a not-so-great "tab" system). However, in order to get nvim-tree to work, I had to let it totally own the buffer, and that drawer looks for buffers named `NvimTree_`. So, if the user has multiple nvim-tree drawers, and they're all named something like `NvimTree_`, it will get confused.

There's also other not-so-great things about the drawer. The order they're created in determines your window layout. I like a terminal across the whole width of the bottom of my window, and my tree on the side. So I define the tree drawer first, then my terminal drawer, and open them on `VimEnter` in that same order.

I tried to work in some sort of `stretch = true` option for drawers but couldn't figure it out.

**Reply by alex-courtis** [reply]
1. I'm not going to commit to winbuffix until it's stable and present.
2. This sounds great, it doesn't sound like any insurmountable technical problems, just conceptual ones.

We can definitely work with you on integration. Mulitple trees isn't real yet; we will have to come to a decision about buffer names etc.

You can look for `NvimTree` filetype to find all the tree buffers.

Do you want to get back to us when it's ready / somewhat stable? Maybe mention me in an issue in the repo when you create it.

**Reply by mikew** [reply]
I've just gone through a huge refactor and am pretty happy with where it is.

It now supports:

- Floating windows
- Remembers whichever buffers were open in drawers by default
  This was inspired by someone at work, they like to keep a `notes.md` file in their repositories. So now they can open a drawer, call `:edit notes.md`, and they have a nice drawer that will stick around as they navigate
- Removed anything relating to "splits", it now just deals with windows, and potentially creates buffers (because a window needs a buffer)

I did create a "potential usecase" tag, and created an nvim-tree tracking issue: https://github.com/mikew/nvim-drawer/issues/2

FWIW, nvim-tree works fine with nvim-drawer, but there are some oddities mentioned in that issue linked above.

**Reply by alex-courtis** [reply]
Congratulations on publishing!

Could I please ask you to add nvim-drawer to the wiki? I've created an Integrations section: https://github.com/nvim-tree/nvim-tree.lua/wiki/Integrations-And-Extension-Plugins with a placeholder for you.

**Reply by mikew** [reply]
Appreciate that, I'll add it now
